tests/fpgadataflow/test_convert_to_hls_thresholding.py

```diff
@@ -0,0 +1,331 @@
+        NumChannels=NumChannels,
+        PE=pe,
+        numSteps=thresholds.shape[1],
+        inputDataType=input_data_type.name,
+        weightDataType=input_data_type.name,
+        outputDataType=output_data_type.name,
+        numInputVectors=num_input_vecs,
+        activation_bias=activation_bias,
+    )
+    graph = helper.make_graph(
+        nodes=[Thresholding_node],
+        name="thresholding_graph",
+        inputs=[inp],
+        outputs=[outp],
+    )
+
+    model = helper.make_model(graph, producer_name="thresholding-model")
+    model = ModelWrapper(model)
+
+    model.set_tensor_datatype("inp", input_data_type)
+    model.set_tensor_datatype("outp", output_data_type)
+
+    model.set_tensor_datatype("thresh", input_data_type)
+    model.set_initializer("thresh", thresholds)
+    return model
+
+
+def make_single_multithresholding_modelwrapper(
+    thresholds,
+    pe,
+    input_data_type,
+    output_data_type,
+    activation_bias,
+    num_input_vecs,
+):
+    NumChannels = thresholds.shape[0]
+
+    inp = helper.make_tensor_value_info(
+        "inp", TensorProto.FLOAT, num_input_vecs + [NumChannels]
+    )
+    outp = helper.make_tensor_value_info(
+        "outp", TensorProto.FLOAT, num_input_vecs + [NumChannels]
+    )
+
+    node_inp_list = ["inp", "thresh"]
+
+    Multithresholding_node = helper.make_node(
+        "MultiThreshold",
+        node_inp_list,
+        ["outp"],
+        domain="qonnx.custom_op.general",
+        out_dtype=output_data_type.name,
+        out_bias=float(activation_bias),
+        out_scale=1.0,
+    )
+
+    graph = helper.make_graph(
+        nodes=[Multithresholding_node],
+        name="multithresholding_graph",
+        inputs=[inp],
+        outputs=[outp],
+    )
+
+    model = helper.make_model(graph, producer_name="multithresholding-model")
+    model = ModelWrapper(model)
+    model = model.transform(InferShapes())
+    model = model.transform(InferDataTypes())
+    model = model.transform(GiveUniqueNodeNames())
+
+    model.set_tensor_datatype("inp", input_data_type)
+    model.set_tensor_datatype("outp", output_data_type)
+
+    model.set_tensor_datatype("thresh", input_data_type)
+    model.set_initializer("thresh", thresholds)
+    return model
+
+
+@pytest.mark.parametrize("activation", [DataType["INT4"], DataType["BIPOLAR"]])
+@pytest.mark.parametrize("input_data_type", [DataType["INT16"], DataType["UINT16"]])
+@pytest.mark.parametrize("fold", [-1, 1, 2])
+@pytest.mark.parametrize("num_input_channels", [16])
+@pytest.mark.parametrize("mem_mode", ["decoupled", "const"])
+@pytest.mark.fpgadataflow
+@pytest.mark.vivado
+def test_convert_to_hls_tbs_rtl_variant(
+    activation, input_data_type, fold, num_input_channels, mem_mode
+):
+    # Handle inputs to the test
+    pe = generate_pe_value(fold, num_input_channels)
+    num_steps = activation.get_num_possible_values() - 1
+
+    # Cppsim is not supported for this node (as it is an RTL node)
+    if mem_mode == "const":
```

Comment:
Because of this comment - https://github.com/Xilinx/finn/pull/687/#discussion_r1046014746
I'll delete make_single_thresholding_binary_search_modelwrapper from this convert_to_hls module - and import the function instead.
The second bullet contradicts the link in this comment - which is preferred?

Reply:
Ignore previous comment - I misunderstood.

Removed duplication for 'make_single_thresholding_binary_search_modelwrapper' - see https://github.com/Xilinx/finn/pull/687/commits/bdfa6cb97096680247b6648edf20d4c519dcad16

mem_mode also set to decoupled and assert now in place - see https://github.com/Xilinx/finn/pull/687/commits/6809351c5210c87a199e8b4167fa54b2dd9a48c8